src/std/net/json-rpc.ss

```diff
@@ -35,30 +37,31 @@
   (only-in :std/text/json trivial-json-object->class JSON json-symbolic-keys
            bytes->json-object json-object->bytes json-object->string))
 
-(defstruct (json-rpc-error <error>) ()
-;;  (code    ;; SInt16
-;;   message ;; String
-;;   data)   ;; (Maybe Bytes)
-  transparent: #t constructor: :init!)
-(defmethod {:init! json-rpc-error}
+(deferror-class (JSON-RPCError IOError) () json-rpc-error?
+  ;;  (code    ;; SInt16
+  ;;   message ;; String
+  ;;   data)   ;; (Maybe Bytes)
```

Comment:
Shouldn't we indeed have code and data be slots in addition to being in irritants? and have message be the regular message slot?

Reply:
absolutely, can do that.